harness/determined/pytorch/_pytorch_context.py

```diff
@@ -168,16 +170,8 @@ def wrap_model(self, model: torch.nn.Module) -> torch.nn.Module:
             check.false(self._use_apex, "Must call wrap_model() before configure_apex_amp.")
 
             model = model.to(self.device)
-            if not self.distributed.size > 1 and self.n_gpus > 1:
```

Comment:
what happens if somebody sets slots_per_trial = 1?

Reply:
they should be able to run torch DDP on a single machine single-slot. are you saying we should have some check to prevent them from doing this?

Reply:
so as-written, the user story is:
- if you want to run vanilla pytorch, use horovod backend with a single-slot trial
- if you want horovod dtrain, use horovod backend with a multi-slot trial
- if you want DDP dtrain, use torch.distributed backend

It seems weird to me that in the single-slot case, we would special-case horovod but not DDP.

Reply:
added single-slot detection for DDP. do you think an internal argument like `--autohorovod` is necessary in this case?